src/GraphQL/Utilities/DirectiveVisitorSelector.cs

```diff
@@ -1,56 +0,0 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using GraphQL.Types;
-using GraphQLParser.AST;
-
-namespace GraphQL.Utilities
-{
-    public class DirectiveVisitorSelector : IVisitorSelector
-    {
-        private readonly Dictionary<string, Type> _directiveVisitors;
-        private readonly Func<Type, SchemaDirectiveVisitor> _typeResolver;
-
-        public DirectiveVisitorSelector(
-            Dictionary<string, Type> directiveVisitors,
-            Func<Type, SchemaDirectiveVisitor> typeResolver)
-        {
-            _directiveVisitors = directiveVisitors;
-            _typeResolver = typeResolver;
-        }
-
-        public IEnumerable<ISchemaNodeVisitor> Select(object node)
-        {
-            if (node is IProvideMetadata meta && meta.GetAstType<IHasDirectivesNode>() is IHasDirectivesNode ast)
-            {
-                if (ast.Directives != null)
-                {
-                    foreach (var visitor in BuildVisitors(ast.Directives))
-                    {
-                        yield return visitor;
-                    }
-                }
-            }
-        }
-
-        private IEnumerable<ISchemaNodeVisitor> BuildVisitors(List<GraphQLDirective> directives)
-        {
-            foreach (var dir in directives)
-            {
-                var name = (string)dir.Name.Value;
-                if (_directiveVisitors.TryGetValue(name, out var type))
-                {
-                    var visitor = _typeResolver(type);
-
-                    if (visitor.Name != name)
-                        throw new InvalidOperationException($"SchemaDirectiveVisitor '{visitor.GetType().Name}' has '{visitor.Name}' name but registered in {nameof(SchemaBuilder)} as '{name}'. Names must match.");
-
-                    if (dir.Arguments != null)
-                        visitor.Arguments = dir.Arguments.ToDictionary(x => (string)x.Name.Value, x => x.Value.ToValue());
```

Comment:
Important! Design flaw! The arguments were set each time, overwriting the previous values. This was a mistake I pointed out a long time ago when I was just starting out #1451. Now directive visitors (well, all visitors) are stateless.